### PR TITLE
Add section on reserved keywords

### LINKED ISSUE #146: Add section on warning for reserved words in JSON schema versions
such as the `$id` and `$schema` properties

add comments on inlining and relative de-referencing (`$ref`) - provide guidance on using inlining

### PR DESCRIPTION
**Pull request by thereisnogabe**
fixes #146

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-json-schema/pull/161.html" title="Last updated on Jun 12, 2023, 3:51 PM UTC (c79afe5)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-json-schema/161/d614bf8...c79afe5.html" title="Last updated on Jun 12, 2023, 3:51 PM UTC (c79afe5)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -375,13 +375,64 @@ <h2>JSON Schema Specifications</h2>
...
             Write considerations for normatively referencing JSON Schema.
           </p>
-          <p class="issue" data-number="142">
-            Write section on using different representations of JSON Schema, such as YAML.
-          </p>
-          <p class="issue" data-number="146">
-            Write section on implementation considerations, such as using <code>name</code>, <code>description</code> and
-            reserved keywords such as <code>$id</code> and <code>$schema</code>.
-          </p>
+          <section class="normative">
```
Comment (OR13) [inline index.html:378]: normative section

File: index.html [hunk 2]
```diff
@@ -375,13 +375,64 @@ <h2>JSON Schema Specifications</h2>
...
+                  value in the <code>credentialSchema</code> object of a <a>verifiable credential</a>, and
+                  that the value of the <code>$id</code> is a network-adderssable URL.
+               </p>
+            </section>
+            <section>
+               <h4>$schema</h4>
+               <p>
+                  Across JSON Schema specifications the <code>$schema</code> keyword identifies a JSON Schema
+                  providing the feature set for a given JSON Schema specification. This property MUST be present
+                  in each schema. For example, when constructing a schema for <a href="https://datatracker.ietf.org/doc/html/draft-bhutton-json-schema-01">Draft 2020-12</a> the value of the <code>$schema</code> identifier MUST
```
Comment (OR13) [inline index.html:411]: downref bad... we can't refer to abandoned drafts from a TR track document.

File: index.html [hunk 3]
```diff
@@ -375,13 +375,64 @@ <h2>JSON Schema Specifications</h2>
...
-          </p>
+          <section class="normative">
+            <h3>Reserved Keywords</h3>
+            <p>
+               JSON Schema specifications reserve certain keywords that hold specific meanings and
+               functions during the processing of JSON Schemas. It is crucial to avoid using conflicting
+               keys when creating JSON Schemas. The specification document for each version of JSON Schema
+               lists these reserved keywords, which can be found in the table provided above.
+
+               In the upcoming sections we list <i>some</i> keywords that possess unique significant and
```
Comment (OR13) [inline index.html:386]:
```suggestion
               In the upcoming sections we list <i>some</i> keywords that possess unique significance and
```

File: index.html [hunk 4]
```diff
@@ -375,13 +375,64 @@ <h2>JSON Schema Specifications</h2>
...
+               should not be used in conflicting ways.
+            </p>
+            <p>
+               Furthermore, we will identify specific keywords that are not explicitly defined by JSON Schema
```
Comment (OR13) [inline index.html:390]:
```suggestion
               Furthermore, we identify specific keywords that are not explicitly defined by JSON Schema
```

File: index.html (hunk 2 above)
Comment (decentralgabe) [reply index.html:411]: will remove

File: index.html [hunk 5]
```diff
@@ -375,13 +375,64 @@ <h2>JSON Schema Specifications</h2>
...
+               <h4>$id</h4>
+               <p>
+                  Across JSON Schema specifications the <code>$id</code> keyword identifies a schema resource
+                  with its canonical [[RFC-6596]] URI. The <code>$id</code> MUST be present and its value
+                  MUST represent a valid URI-reference [[RFC-3986]].
+               </p>
+               <p>
+                  It is RECOMMENDED that the value of the <code>$id</code> property match the <code>id</code>
+                  value in the <code>credentialSchema</code> object of a <a>verifiable credential</a>, and
+                  that the value of the <code>$id</code> is a network-adderssable URL.
```
Comment (OR13) [inline index.html:403]:
```suggestion
                  that the value of the <code>$id</code> is a dereferenceable IRI.
```

Comment (OR13) [reply index.html:403]: ^ this aligns more with the core data model, then the previous text.

**Review by OR13 — CHANGES_REQUESTED** [review]
need to address downref, I suggest just removing the reference.

**Review by andresuribe87 — APPROVED** [review]

File: index.html [hunk 6]
```diff
@@ -105,7 +105,7 @@ <h2>Abstract</h2>
           Verifiable Presentations, and explains the relationships between three parties:
           <i>issuers</i>, <i>holders</i>, and <i>verifiers</i>. Verifiability, extensibility, and semantic
           interoperability are critical pieces of functionality referenced throughout
-          the [VC-DATA-MODEL-2]. This specification provides a mechanism to make use of a Credential Schema in
+          the [[VC-DATA-MODEL-2]. This specification provides a mechanism to make use of a Credential Schema in
```
Comment (TallTed) [inline index.html:108]:
```suggestion
          the [[VC-DATA-MODEL-2]]. This specification provides a mechanism to make use of a Credential Schema in
```

File: index.html [hunk 7]
```diff
@@ -375,13 +375,64 @@ <h2>JSON Schema Specifications</h2>
...
-            reserved keywords such as <code>$id</code> and <code>$schema</code>.
-          </p>
+          <section class="normative">
+            <h3>Reserved Keywords</h3>
+            <p>
+               JSON Schema specifications reserve certain keywords that hold specific meanings and
+               functions during the processing of JSON Schemas. It is crucial to avoid using conflicting
+               keys when creating JSON Schemas. The specification document for each version of JSON Schema
+               lists these reserved keywords, which can be found in the table provided above.
+
```
Comment (TallTed) [inline index.html:385]:
```suggestion
            </p>
            <p>
```

File: index.html [hunk 8]
```diff
@@ -375,13 +375,64 @@ <h2>JSON Schema Specifications</h2>
...
+               In the upcoming sections we list <i>some</i> keywords that possess unique significance and
+               should not be used in conflicting ways.
+            </p>
+            <p>
+               Furthermore, we identify specific keywords that are not explicitly defined by JSON Schema
```
Comment (TallTed) [inline index.html:390]:
```suggestion
               Furthermore, we identify specific keywords, that are not explicitly defined by JSON Schema,
```

File: index.html [hunk 9]
```diff
@@ -375,13 +375,64 @@ <h2>JSON Schema Specifications</h2>
...
+               but are emphasized in this specification to support widespread usage.
+            </p>
+            <section>
+               <h4>$id</h4>
+               <p>
+                  Across JSON Schema specifications the <code>$id</code> keyword identifies a schema resource
```
Comment (TallTed) [inline index.html:396]:
```suggestion
                  Across JSON Schema specifications, the <code>$id</code> keyword identifies a schema resource
```

File: index.html [hunk 10]
```diff
@@ -375,13 +375,64 @@ <h2>JSON Schema Specifications</h2>
...
+               <p>
+                  It is RECOMMENDED that the value of the <code>$id</code> property match the <code>id</code>
+                  value in the <code>credentialSchema</code> object of a <a>verifiable credential</a>, and
+                  that the value of the <code>$id</code> is a dereferenceable IRI.
+               </p>
+            </section>
+            <section>
+               <h4>$schema</h4>
+               <p>
+                  Across JSON Schema specifications the <code>$schema</code> keyword identifies a JSON Schema
```
Comment (TallTed) [inline index.html:409]:
```suggestion
                  Across JSON Schema specifications, the <code>$schema</code> keyword identifies a JSON Schema
```

File: index.html [hunk 11]
```diff
@@ -375,13 +375,64 @@ <h2>JSON Schema Specifications</h2>
...
+               <p>
+                  It is RECOMMENDED that all JSON Schemas include a <code>name</code> property which provides
+                  a human-readable name that describes the schema.
+               </p>
+            </section>
+            <section>
+               <h4>description</h4>
+               <p>
+                  JSON Schemas MAY choose to include an optional <code>description</code> property which provides
+                  a human-readible sentence describing the schema.
```
Comment (TallTed) [inline index.html:426]:
```suggestion
                  a human-readable sentence describing the schema.
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Plus the few inadvertently posted individually.

**Review by OR13 — APPROVED** [review]

**Pull request merged**